### PR TITLE
make event history tx fields optional

### PR DESCRIPTION
**Pull request by zeroXbrock**
fixes this error

Screenshot from 2023-07-10 21-51-58

### CODE REVIEW

File: crates/mev-share-sse/src/types.rs [hunk 1]
```diff
@@ -26,13 +26,13 @@ pub struct Event {
...
-    pub to: Address,
+    pub to: Option<Address>,
     /// 4-byte-function selector
     #[serde(rename = "functionSelector")]
-    pub function_selector: FunctionSelector,
+    pub function_selector: Option<FunctionSelector>,
     /// Calldata of the transaction
     #[serde(rename = "callData")]
-    pub calldata: Bytes,
+    pub calldata: Option<Bytes>,
```
Comment (mattsse) [inline crates/mev-share-sse/src/types.rs:35]: so all of these are optional?

should they be serialized as null or skipped?

if skipped, we need a skip_serialize_if annotation

File: crates/mev-share-sse/src/types.rs [hunk 2]
```diff
@@ -184,7 +184,7 @@ impl<'de> Deserialize<'de> for FunctionSelector {
             return Err(serde::de::Error::custom(format!(
                 "Expected 4 byte function selector: {}",
                 hex_str
-            )))
+            )));
```
Comment (mattsse) [inline crates/mev-share-sse/src/types.rs:187]:
```suggestion
            )))
```

File: crates/mev-share-sse/src/types.rs (hunk 1 above)
Comment (zeroXbrock) [reply crates/mev-share-sse/src/types.rs]: should be skipped, updated

File: crates/mev-share-sse/src/types.rs (hunk 2 above)
Comment (zeroXbrock) [reply crates/mev-share-sse/src/types.rs:190]: lol my lint-on-save wouldn't let me undo that

**Review by mattsse — APPROVED** [review]

**Pull request merged**